rules/certificates.bzl

```diff
@@ -0,0 +1,88 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("//rules:rv.bzl", "rv_rule")
+load("//rules/opentitan:toolchain.bzl", "LOCALTOOLS_TOOLCHAIN")
+
+def _certificate_codegen_impl(ctx):
+    tc = ctx.toolchains[LOCALTOOLS_TOOLCHAIN]
+
+    basename = paths.replace_extension(ctx.file.template.path, "")
+
+    out_c = ctx.actions.declare_file("{}.c".format(basename))
+    out_h = ctx.actions.declare_file("{}.h".format(basename))
+    outputs = [out_c, out_h]
+    ctx.actions.run(
+        outputs = outputs,
+        inputs = [
+            ctx.file.template,
+        ],
+        arguments = [
+            "--rcfile=",
+            "--quiet",
+            "certificate",
+            "codegen",
+            "--template={}".format(ctx.file.template.path),
+            "--output-c={}".format(out_c.path),
+            "--output-h={}".format(out_h.path),
+        ],
+        executable = tc.tools.opentitantool,
```

Comment:
we probably also want to hit the generated C files with clang format after they are generated?

Reply:
I will look into it.